### DISCUSSION TITLE
IIIF endpoint

### DISCUSSION THREAD

**Discussion by HemboPagi** [directus/directus#15495] [Feature Requests]
Hello

I am working with the dataset with lots of images. Also, some images will be displayed from other system via IIIF (iiif.io) viewer. I am wondering if there are any plans to implement IIIF image/presentation API as one possible output for media files? Or maybe someone has already implementet it as an extension?

All the best
Hembo

**Comment by benhaynes** [comment]
Not sure how widely adopted this IIIF spec is, but I think it could either be added to core (if truly a standard), or be a API Endpoint extension that gets added to the (upcoming) marketplace. 👍

**Reply by HemboPagi** [reply]
IIIF is widely adopted in memory institutions (libraries, other research institutions). I know one big project runnig their databases via Directus but for files (images) they use file repository with IIIF support. I will need to decide in few week time how I continue with our (research) database, as we do not have resources for the full IIIF extenison deveopment.

**Reply by benhaynes** [reply]
If it's important and time dependent, you could reach out via our website's live chat to see about sponsoring it. 👍

**Comment by rijkvanzanten** [comment]
@HemboPagi IIIF seems to have multiple API specifications. Are you looking for just the "Image API" or also the other ones? https://iiif.io

**Reply by HemboPagi** [reply]
Hello

Image API is the starting point, but for the user to be able to use IIIF tools also presentation API is required. What i did with an other (WoerdPress based) project, was that i used image server (IIP Image) with IIIF image API support and wrote my own Presentation manifest endpoint for WordPress. And, as in Directus all the images by default are controlled by user access, then Authentication API could be alos useful. In our project it is not needed tho.

With our current project we could get around actually without Image API I belive, as the viewer (Mirador) we use is able to handle also full size images (Static example where the image comes form Directus setup: https://gist.githubusercontent.com/HemboPagi/79aa821270fbd2fcde1ef388e6f22008/raw/83e8a0b25ad1236b43f6b6a8141a1765e94c8ecc/manifest.json and via the viewer: https://projectmirador.org/embed/?iiif-content=https://gist.githubusercontent.com/HemboPagi/79aa821270fbd2fcde1ef388e6f22008/raw/83e8a0b25ad1236b43f6b6a8141a1765e94c8ecc/manifest.json).  For the user it means tho that the full image will be dowloaded at once.

**Comment by HemboPagi** [comment]
Had a look on Sharp. it supports IIIF info.json output : https://sharp.pixelplumbing.com/api-output#tile

**Comment by HemboPagi** [comment]
Hello, I am soon ready to make public the first version of the IIIF extensin for Directus. Currently the setup is done via extra collection, as had not time to figure out how to add config options directly to project setup page. Is it possible?

Also, as I mentioned above, Sharp outputs  also IIIF info for tiled images - this is missing from my extenison. Can you confirm that Directus is capable of producein tiled images as well? Or maybe it is an small update for you?

An example manifest output : https://db.dl.tlu.ee/iiif/manifest/parchment/14060

This can be viewd via IIIF viewer, for example https://projectmirador.org/embed/?iiif-content=https://db.dl.tlu.ee/iiif/manifest/parchment/14060

**Reply by rijkvanzanten** [reply]
Awesome!

> [...] to add config options directly to project setup page. Is it possible?

No, there's currently no built-in way for extensions to register Settings.

> Can you confirm that Directus is capable of producein tiled images as well?

Could you elaborate on what you mean by "produce" tiled images?

**Reply by HemboPagi** [reply]
So what would be the best way to deal with extension settings then? 

As i understood Directus uses Sharp for image manipulation. So Sharp has an option for tiled images (deep zoom - image pyramid) https://sharp.pixelplumbing.com/api-output#tile

**Reply by HemboPagi** [reply]
Also, just to highlight the importance of tiled/zoomable images: soon I will get an archive with hudreds of TIF files (40-50 BM each) and in our current IIIF setup TIF is shown as jpeg and then the full image is sent tu user. So if we can provide zoomable image via IIIF image service (as Sharp outputs it) then user needs to dowload a fraction of the data.

**Reply by paescuj** [reply]
> So what would be the best way to deal with extension settings then?

I think using a [singleton collection](https://docs.directus.io/getting-started/glossary.html#singleton) is the current way to go as you seem to be using already.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
